### ISSUE TITLE
Some List* endpoints don't take ListOptions (but they should).

### ISSUE THREAD

**Issue by shurcooL** [google/go-github#636]
Most (but not all) list endpoints support pagination, and take `opt *ListOptions` as the last parameter (or another specific Options type that embeds `ListOptions`).

Some `List*` endpoints do not take a `*ListOptions` or equivalent parameter, and therefore it's not possible to perform pagination.

It's a breaking API change, but we should fix it sooner rather than later.

One such case was the `PullRequestsService.ListReviews` method, reported by @sminnee in #635.

Additional cases I found just by looking briefly at the code include [`PullRequestsService.ListReviewComments`](https://gotools.org/github.com/google/go-github/github?rev=c03cf196876fc0b7cf8db2516848536778ea4e29#PullRequestsService.ListReviewComments) and [`AuthorizationsService.ListGrants`](https://gotools.org/github.com/google/go-github/github?rev=c03cf196876fc0b7cf8db2516848536778ea4e29#AuthorizationsService.ListGrants). There are probably more, I haven't done an exhaustive search.

The task is twofold:

1. Identify all such problematic endpoints and list them here.
2. Send a PR similar to #635 to fix them and bump `libraryVersion` up by one.

Also, we should be more mindful of this potential pitfall when doing reviews in the future. /cc @gmlewis

**Comment by sminnee** [comment]
Ok so I went through all the "List" functions and these are the ones that don't have options parameters:

 * [ ] activity.go: ListFeeds
 * [ ] authorizations.go: ListGrants
 * [ ] gists.go: ListCommits
 * [ ] gitignore.go: List
 * [ ] licenses.go: List
 * [ ] misc.go: ListEmojis
 * [ ] misc.go: ListServiceHooks
 * [ ] pulls_reviewers.go: ListReviewers
 * [ ] pulls_reviews.go ListReviewComments
 * [ ] repos.go: ListLanguages
 * [ ] repos.go: ListRequiredStatusChecksContexts
 * [ ] repos_pages.go: ListPagesBuilds
 * [ ] repos_stats.go: ListContributorsStats
 * [ ] repos_stats.go: ListCommitActivity
 * [ ] repos_stats.go: ListCodeFrequency
 * [ ] repos_stats.go: ListParticipation
 * [ ] repos_stats.go: ListPunchCard
 * [ ] repos_traffic.go: ListTrafficReferrers
 * [ ] repos_traffic.go: ListTrafficPaths
 * [ ] users.go: ListInvitations

**Comment by sminnee** [comment]
The indication from https://developer.github.com/v3/#pagination is that every list-based feed will paginate when more than 30 items are returned. So I would think that adding ListOptions to all of these is the best idea.

**Comment by shurcooL** [comment]
> I haven't yet validated whether the API provides pagination on each of these.

It's important that we do this.

For example, `ActivityService.ListFeeds` in `activity.go` should _not_ have `ListOptions`, because the endpoint does not support pagination:

https://developer.github.com/v3/activity/feeds/#list-feeds

Notice that it returns a single JSON object, not array, and there's no "Link" headers with pagination under "Status: 200 OK" text:

![image](https://cloud.githubusercontent.com/assets/1924134/25841220/4a56520a-346d-11e7-80a8-938faff2a084.png)

It's just an endpoint that happens to have "List" in its name.

Endpoints that actually support pagination look like this:

![image](https://cloud.githubusercontent.com/assets/1924134/25841300/8d279e68-346d-11e7-9a94-5abb4510b2bf.png)

Notice they return a JSON array of objects, and there are Link headers with pagination under "Status: 200 OK".

**Comment by sminnee** [comment]
I've had a look into this: I think that the `Link` heading is only returned if there are more than 1 page of results — it is not returned on a single-page response that supports pagination in cases where there are more items. 

So, I've found a better way to check is to put "?page=2" on the end of the URL and see if the output changes.

**Comment by sminnee** [comment]
OK: I've updated my previous comment break the list out into pagination / non.

**Comment by sminnee** [comment]
I've now expanded the scope of #635 to be a complete fix to this. You should be able to close this issue once that is merged.

**Comment by shurcooL** [comment]
> I don't have a place to test this but I assume it's got pagation.
>
> - [ ] repos_pages.go: ListPagesBuilds

It's [documented](https://cloud.githubusercontent.com/assets/1924134/25931360/e129c280-35d9-11e7-849c-ef36122a1150.png) to have it, yes.

> Ok so I went through all the "List" functions and these are the ones that don't have options parameters.

I've verified, and everything looks good, except "licenses.go: List".

It's [documented](https://developer.github.com/v3/licenses/#list-all-licenses) not to have pagination support:

![image](https://cloud.githubusercontent.com/assets/1924134/25931389/06532394-35da-11e7-8fe9-580c5d261af0.png)

When I did:

```
$ curl -H 'Accept: application/vnd.github.drax-preview+json' -i https://api.github.com/licenses | grep Link
```

The Link header wasn't set.

However, I saw that doing:

```
$ curl -H 'Accept: application/vnd.github.drax-preview+json' -i https://api.github.com/licenses?page=2
```

```
$ curl -H 'Accept: application/vnd.github.drax-preview+json' -i https://api.github.com/licenses?per_page=1
```

Actually works and sets Link header then.

I don't think we should add it. It's likely that the endpoint supports pagination unintentionally, which could be a bug that GitHub will fix. There's only ~8 licenses, chances are they won't grow exponentially, so there's not going to be need for pagination.

All others look good. Thanks for doing the work!

**Comment by shurcooL** [comment]
Actually, `UsersService.ListInvitations` is documented not to have pagination either:

https://developer.github.com/v3/repos/invitations/#list-a-users-repository-invitations

But given the nature of repository invitations, I expect that endpoint should absolutely support pagination. I suspect it's a bug in their API documentation.

Would you want to send them an email at support@github.com to get clarification and get it fixed?

**Comment by sminnee** [comment]
OK I've emailed github about this.

Looks like licenses was an clerical error on my part, I'll remove from the PR.

**Comment by shurcooL** [comment]
We got a reply from GitHub, see https://github.com/google/go-github/pull/635#issuecomment-300743066.

It confirms that `UsersService.ListInvitations` endpoint should also support pagination, as we expected.

**Comment by shurcooL** [comment]
This issue was fully resolved by #635.
